python/packages/autogen-agentchat/src/autogen_agentchat/agents/_assistant_agent.py

```diff
@@ -814,14 +813,13 @@ async def _add_messages_to_context(
         messages: Sequence[ChatMessage],
     ) -> None:
         """
-        Add incoming user (and possibly handoff) messages to the model context.
+        Add incoming messages to the model context.
         """
         for msg in messages:
             if isinstance(msg, HandoffMessage):
-                # Add handoff context to the model context.
-                for context_msg in msg.context:
-                    await model_context.add_message(context_msg)
-            await model_context.add_message(UserMessage(content=msg.content, source=msg.source))
+                for llm_msg in msg.context:
+                    await model_context.add_message(llm_msg)
+            await model_context.add_message(msg.content_to_model_message())
```

Comment:
How do you know if it is meant to be a user or assistant message?

Reply:
Because a ChatMessage is meant to be between agent to agent, and as external data it has to be user message not assistant message. 

This prevents accidently updating an agent's model context with assistant message from a different agent.